frontend/src/Update.elm

```diff
@@ -0,0 +1,232 @@
+-}
+route : Navigation.Location -> Maybe Route
+route location =
+    let
+        routeParser : UrlParser.Parser (Route -> a) a
+        routeParser =
+            UrlParser.oneOf
+                [ UrlParser.map Home top
+                , UrlParser.map AttributionRoute (s "attribution")
+                , UrlParser.map ProfilingRoute (s "profiling")
+                ]
+    in
+        UrlParser.parsePath routeParser location
+
+
+initialState : Navigation.Location -> ( Model, Cmd Msg )
+initialState location =
+    let
+        ( navbarState, navbarCmd ) =
+            Navbar.initialState NavbarMsg
+
+        defaultAttribution =
+            { knownAuthorMode = PasteText
+            , knownAuthorText = ""
+            , unknownAuthorMode = PasteText
+            , unknownAuthorText = ""
+            , result = Nothing
+            , language = EN
+            }
+
+        defaultProfiling =
+            { mode = PasteText
+            , text = fillerText1
+            , result = Just { gender = M, age = 20 }
+            }
+
+        defaultRoute =
+            route location
+                |> Maybe.withDefault Home
+    in
+        ( { route = defaultRoute
+          , navbarState = navbarState
+          , profiling = defaultProfiling
+          , attribution = defaultAttribution
+          }
+        , navbarCmd
+        )
+
+
+toggleInputMode : InputMode -> InputMode
+toggleInputMode mode =
+    case mode of
+        FileUpload ->
+            PasteText
+
+        PasteText ->
+            FileUpload
+
+
+{-| How our model should change when a message comes in
+
+* NoOp, does nothing
+* NavBarMsg, updates highlight in the navigation bar
+* ChangeRoute, changes the route - the currently displayed page
+* UrlChange, does nothing, see comment
+* AttributionMsg, nested update on the attribution
+* ProfilingMsg, nested update on the profiling
+-}
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        NoOp ->
+            ( model, Cmd.none )
+
+        NavbarMsg state ->
+            ( { model | navbarState = state }, Cmd.none )
+
+        ChangeRoute newRoute ->
+            if newRoute == model.route then
+                ( model, Cmd.none )
+            else
+                let
+                    _ =
+                        Debug.log "new route" newRoute
+
+                    newUrl =
+                        case newRoute of
+                            Home ->
+                                "/"
+
+                            AttributionRoute ->
+                                "attribution"
+
+                            ProfilingRoute ->
+                                "profiling"
+                in
+                    ( { model | route = newRoute }
+                      -- update the url to represent the currently displayed page.
+                    , Navigation.newUrl newUrl
+                    )
+
+        UrlChange location ->
```

Comment:
well it is used in `Main.elm`, because we have to do something with the signal. 

```elm
main : Program Never Model Msg
main =
    Navigation.program UrlChange
        { update = Update.update
        , view = View.view
        , init = Update.initialState
        , subscriptions = subscriptions
        }
```
the alternative is 
```elm
main : Program Never Model Msg
main =
    Navigation.program (\_ -> NoOp)
        { update = Update.update
        , view = View.view
        , init = Update.initialState
        , subscriptions = subscriptions
        }
```
We discussed this, and decided that the descriptive name is better, even if it has no effect.